src/indexer/services/substrateIndexer/substrateIndexer.ts

```diff
@@ -46,12 +66,66 @@ export class SubstrateIndexer {
           toBlock = fromBlock + this.pastEventsQueryInterval
         }
 
-        await this.saveDataToDb(this.domain.id, toBlock.toString())
+        const blockHash = await this.provider.rpc.chain.getBlockHash(fromBlock)
+        const signedBlock = await this.provider.rpc.chain.getBlock(blockHash)
+        const at = await this.provider.at(blockHash)
+        const timestamp = Number((await at.query.timestamp.now()).toString())
+        const allRecords = await at.query.system.events()
+
+        // we get the proposal execution events - ts-ignore because of allRecords
+        // @ts-ignore
+        const proposalExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.ProposalExecution, allRecords) as Array<ProposalExecutionEvent>
+
+        // we get the deposit events - ts-ignore because of allRecords
+        // @ts-ignore
+        const depositEvents = getSubstrateEvents(SygmaPalleteEvents.Deposit, allRecords) as Array<DepositEvent>
+
+        // we get the index of the section in the extrinsic
+        const sectionIndex = signedBlock.block.extrinsics.findIndex(ex => ex.method.section === "sygmaBridge")
+
+        // this is our identifier for the tx
+        const txIdentifier = `${fromBlock}-${sectionIndex}` //this is like the txHash but for the substrate
+
+        if (proposalExecutionEvents.length) {
+          proposalExecutionEvents.forEach((proposalExecutionEvent: ProposalExecutionEvent) => {
+            const { data } = (proposalExecutionEvent.event as any).toHuman()
+
+            const { originDomainId, depositNonce } = data
+
+            this.saveProposalExecutionToDb(this.domain.id, fromBlock.toString(), {
+              originDomainId,
+              depositNonce: depositNonce,
+              txIdentifier,
+              blockNumber: `${fromBlock}`,
+              timestamp
+            })
+          })
+        } else if (depositEvents.length) {
+          depositEvents.forEach((depositEvent: DepositEvent) => {
+            const { data } = (depositEvent.event as any).toHuman()
+
+            const { destDomainId, resourceId, depositNonce, sender, transferType, depositData, handlerResponse } = data
+
+            this.saveDepositToDb(this.domain.id, fromBlock.toString(), {
+              destDomainId,
+              resourceId,
+              depositNonce: depositNonce,
+              sender,
+              transferType,
+              depositData,
+              handlerResponse,
+              txIdentifier,
+              blockNumber: `${fromBlock}`,
+              timestamp
+            })
+          })
+        }
```

Comment:
You have failed handler execution on subsribe to new head and not here.

Reply:
added